aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -891,89 +875,44 @@ module aptos_framework::coin {
     ///
     /// Note: This bypasses CoinStore::frozen -- coins within a frozen CoinStore can be burned.
     public fun burn_from<CoinType>(
-        account_addr: address,
-        amount: u64,
-        burn_cap: &BurnCapability<CoinType>
-    ) acquires CoinInfo, CoinStore, CoinConversionMap, PairedFungibleAssetRefs {
+        account_addr: address, amount: u64, burn_cap: &BurnCapability<CoinType>
+    ) acquires CoinInfo, CoinConversionMap, PairedFungibleAssetRefs {
         // Skip burning if amount is zero. This shouldn't error out as it's called as part of transaction fee burning.
         if (amount == 0) { return };
-
-        let (coin_amount_to_burn, fa_amount_to_burn) =
-            calculate_amount_to_withdraw<CoinType>(account_addr, amount);
-        if (coin_amount_to_burn > 0) {
-            let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-            let coin_to_burn = extract(&mut coin_store.coin, coin_amount_to_burn);
-            burn(coin_to_burn, burn_cap);
-        };
-        if (fa_amount_to_burn > 0) {
-            fungible_asset::burn_from(
-                borrow_paired_burn_ref(burn_cap),
-                primary_fungible_store::primary_store(
-                    account_addr,
-                    option::destroy_some(paired_metadata<CoinType>())
-                ),
-                fa_amount_to_burn
-            );
-        };
+        fungible_asset::burn_from(
+            borrow_paired_burn_ref(burn_cap),
+            primary_fungible_store::primary_store(
+                account_addr, ensure_paired_metadata<CoinType>()
+            ),
+            amount
+        );
```

Comment:
You don't want the if statement anymore?  I guess it doesn't really matter.

Reply:
cuz nobody will have coinstore anymore